core/src/test/scala/quasar/fs/WriteFileSpec.scala

```
@@ -114,13 +120,19 @@ class WriteFileSpec extends Specification with ScalaCheck with FileSystemFixture
     }
 
     "save should fail and write nothing when source fails" ! prop {
-      (f: AFile, xs: Vector[Data]) =>
+      (f: AFile, xs: Vector[Data], ys: Vector[Data]) =>
 
-      // TODO: handle the error, and check that the file is not written
+      val src = xs.toProcess ++
+                Process.fail(new RuntimeException("SRCFAIL")) ++
+                ys.toProcess
 
-      val p = write.save(f, xs.toProcess ++ Process.fail(new RuntimeException("source failed")))
-      MemTask.runLogEmpty(p).attemptRun must beLeftDisjunction
-    }.pendingUntilFixed("SD-1544")
+      val p = write.save(f, src).drain onFailure {
+        case err if err.getMessage == "SRCFAIL" => read.scanAll(f)
+        case err                                => Process.fail(err)
+      }
+
+      MemTask.runLogEmpty(p).run.toEither must beLeft(pathErr(pathNotFound(f)))
```

Ah, I understand better now. I think this test would be better if there was some arbitrary data already there and we made sure it was still there as from a black box perspective it seems believable that the operation would wipe existing data and then leave it empty if the source were to fail.